bn-apps/cordapp-updates-distribution/corda-updates-core/src/main/kotlin/net/corda/businessnetworks/cordaupdates/core/CordappSyncer.kt

```diff
@@ -0,0 +1,122 @@
+package net.corda.businessnetworks.cordaupdates.core
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import org.eclipse.aether.RepositoryListener
+import org.eclipse.aether.artifact.DefaultArtifact
+import org.eclipse.aether.transfer.TransferListener
+import java.io.File
+
+/**
+ * A wrapper around [CordaMavenResolver] that can sync different CorDapps from different remote repositories via single method invocation
+ *
+ * @syncerCong syncer configuration
+ * @repositoryListener repository listener that will be passed to an underlying [CordaMavenResolver] instance
+ * @transferListener transfer listener that will be passed to an underlying [CordaMavenResolver] instance
+ */
+class CordappSyncer(private val syncerConf : SyncerConfiguration,
+                    private val repositoryListener : RepositoryListener? = null,
+                    private val transferListener : TransferListener? = null) {
+
+    /**
+     * Downloads locally missing CorDapps versions from the remote repositories.
+     *
+     * @coordinatesWithoutVersion optional maven coordinates of a CorDapp, without a version i.e. "net.corda:corda-finance".
+     *              If not provided, then all missing versions of all CorDapps defined in the [syncerConf] will be downloaded.
+     *              If provided, then [syncerConf] is expected to contain a configured [CordappSource] for the CorDapp coordinates.
+     * @additionalConfigurationProperties additional parameters to be passed to [CordaMavenResolver]
+     * @return a list of CorDapp metadatas that have been resolved from local / remote repositories
+     *
+     * @throws [CordappSourceNotFoundException] if CorDapp source has not been found
+     */
+    fun syncCordapps(coordinatesWithoutVersion : String? = null, additionalConfigurationProperties : Map<String, Any> = mapOf()) : List<ArtifactMetadata> {
+        return if (coordinatesWithoutVersion == null) {
+            syncerConf.cordappSources.flatMap { syncTask ->
+                val resolver = CordaMavenResolver.create(syncerConf, syncTask, repositoryListener = repositoryListener, transferListener = transferListener)
+                syncTask.cordapps.map {
```

Comment:
This is actually just a for loop in disguise: use `for (app in syncTask.cordapps)`

Reply:
If I use the for loop then I won't be able to return the result back to the `flatMap`

Reply:
Ah, you're right, sorry.